lib/facility.rb

```diff
@@ -1,14 +1,97 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :phone,
+              :address,
+              :services,
+              :collected_fees,
+              :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @phone = facility_details[:phone]
+    @address = facility_details[:address]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if services.include?("Vehicle Registration")
+      @registered_vehicles << vehicle
+      vehicle.set_date
+      if  vehicle.antique? # If vehicle is older than 25 years, it is :antique
+        @collected_fees += 25 # Add $25 to @collected_fees
+        vehicle.set_plate(:antique)
+      elsif vehicle.electric_vehicle? # Elsif vehicle = EV
+        @collected_fees += 200    # Add $200 to @collected_fees
+        vehicle.set_plate(:ev)
+      else
+        @collected_fees += 100 # Else add $100 to @collected_fees
+        vehicle.set_plate(:regular)
+      end
+    end
+  end
+
+
+
+# Administer a written test
+# A written test can only be administered to registrants with a permit and who are at least 16 years of age
+
+  def administer_written_test(registrant)
+    if services.include?("Written Test")
+      if registrant.permit? && registrant.age >= 16
+        registrant.license_data[:written] = true
+      else
+        false
+      end
+    else
+      false
+    end
+  end
+
+  def administer_road_test(registrant)
+    if services.include?("Road Test") 
+      registrant.permit? && registrant.age >= 16
```

Comment:
this line is correct, but it's not connected to anything. Would need another `if` in front of it, or a `&&` to tie it to the line above.